_posts/blog/2016-09-13-code-reviews.md

```diff
@@ -0,0 +1,70 @@
+
+
+# Do not enforce consensus.
+
+Engineers often tend to forget that there is more than one valid opinion in a lot of cases. Not everything is determined to false and true.
+So if the reviewer has communicated his opinion appropriate and was not able to convince the reviewee, the code review is the wrong forum to let discussion escalate and enforce consensus. If the code is robust and fulfills the requirements, let it pass the review and try to move the topic out of the situation. You may have tech meetings or other forums in which you can discuss the topic (ideally in general and not by specific example) with a broader audience.
+Letting the topic lie for a week can also help. Often times you realize that consensus isnt important at all. That not only there are more valid opinions than yours. But also that your team maybe can afford to let both opinions stand.
+
+# Do not point at rules but give explanations
+
+Rules and processes kill creativity. You want to give as much freedom to everyone as possible without hurting someone else (thats part of the german constitution by the way ;)).
+Developers develop processes and methods. They align on conventions. And other developers violate those. Often times even consceniously. How to handle those violations?
+Often times now is the time when the reviewer changes from consultant to finger-pointing judge (and hangman) who points at the wiki in which he documented the rule three years ago (without ever discussing it properly). Usually the tone gets rough (catch some breath, adjust your mindset...).
+This behaviour is bad because creative people have the healthy habit to question rules if they are in their way. If the result is that the rule does not create value (but supresses value-creation) in the case the engineer, striving to achieve a goal, will try to navigate around it. In such a case: question the rule, not the violator (in fact violating/questioning rules is a healthy habit).
+So back to the situation. State your feedback like this: "at some point we decided to approach things like this like that because we were convinced thats better because of reason a and b. dont you agree? shall we re-evaluate?"
+Usually the topic is done now. The reviewee will probably accept the suggestion because the reviewer acted in a consulting role and took the reviewee serious, asked for his opinion.
+The reviewee will supposely not feel that the rule harms him so badly any more and follow it. Or a healthy discussion arises that in the end might change something for the better for the whole team.
+As a general rule of thumb: do not define too many rules and processes. And in a review: start with understanding the requirements, thinking about potential solutions also from a high-level/architectural/design-perspective. How would you have solved the problem? Then look at the solution. Find real bugs. Only beginners focus on cosmetics (and usually miss real flaws).
```

Comment:
You're adding "How would you have solved the problem?" but you don't connect it at all with the actual review. 
How about:
> Think about potential solutions also from a high-level/architectural/design-perspective and how they can improve the solution that's being reviewed.